core/src/main/java/de/danoeh/antennapod/core/sync/SyncService.java

```diff
@@ -69,29 +81,38 @@
     private static final String WORK_ID_SYNC = "SyncServiceWorkId";
     private static final ReentrantLock lock = new ReentrantLock();
 
-    private ISyncService syncServiceImpl;
+    private final Map<String, ISyncService> syncServices = new HashMap<>();
 
     public SyncService(@NonNull Context context, @NonNull WorkerParameters params) {
         super(context, params);
+        syncServices.put(SYNC_PROVIDER_CHOICE_GPODDER_NET, new GpodnetService(AntennapodHttpClient.getHttpClient(),
+                GpodnetPreferences.getHosturl(), GpodnetPreferences.getDeviceID(),
+                GpodnetPreferences.getUsername(), GpodnetPreferences.getPassword()));
+
+        syncServices.put(
+                SYNC_PROVIDER_CHOICE_NEXTCLOUD,
+                new NextcloudSyncService(getApplicationContext(), ClientConfig.USER_AGENT)
+        );
     }
 
     @Override
     @NonNull
     public Result doWork() {
-        if (!GpodnetPreferences.loggedIn()) {
-            return Result.success();
-        }
-        syncServiceImpl = new GpodnetService(AntennapodHttpClient.getHttpClient(),
-                GpodnetPreferences.getHosturl(), GpodnetPreferences.getDeviceID(),
-                GpodnetPreferences.getUsername(), GpodnetPreferences.getPassword());
         SharedPreferences.Editor prefs = getApplicationContext().getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE)
                 .edit();
+        String selectedService = getSelectedSyncProviderKey(getApplicationContext());
+        ISyncService activeSyncProvider = getActiveSyncProvider(selectedService, this.syncServices);
+
+        return getResultForService(prefs, activeSyncProvider);
```

Comment:
To be honest, I am not really sure why this needs to be split into two different methods

Reply:
Because the method doWork() was too long. The method name of doWork() already hints at that (as a method name should reflect/speak the intention of the implementation).

The new method is easier to grasp  since the selected synchronization service is passed as an argument. Having the logic for resolving the selected service inside the method just adds complexity to the method thus increasing cognitive load reading and understanding the whole method.

Reply:
I extracted the sync provider resolve logic to a separate method. The previous split was not well